### PR TITLE
Removing long (part 7): Various miscellaneous changes

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
SUMMARY: Infrastructure "Remove most uses of long and unsigned long from the code"

#### Purpose of change
My ongoing campaign to remove uses of `long`.

#### Describe the solution
Many assorted small changes.  Mostly they just come down to changing `long` or `unsigned long` to either `int`, `size_t`, `uint32_t`, or `int64_t` as seems appropriate, or suppressing the warnings if `long` is the right choice.  Some other things:
* Use explicit conversion to `bool` on `item_location`.
* Removed a random unnecessary old-style cast.
* Updated a comment.

See the individual commit comments for details.

#### Additional context
I think this is essentially all the remaining uses of `long` caught and dealt with, but there are probably a couple more lurking (there are some uses that my clang-tidy check can't find, for example in code that isn't compiled in this configuration).

### CODE REVIEW

**Comment by kevingranade** [comment]
MINGW and XCode are both unhappy:
```
line_test.cpp: In function 'void line_to_comparison(int)':
line_test.cpp:341:39: error: format '%ld' expects argument of type 'long int', but argument 3 has type 'int64_t {aka long long int}' [-Werror=format=]
                     iterations, diff1 );
                                       ^
line_test.cpp:343:39: error: format '%ld' expects argument of type 'long int', but argument 3 has type 'int64_t {aka long long int}' [-Werror=format=]
                     iterations, diff2 );
                                       ^
```

**Comment by ZhilkinSerg** [comment]
When building under MSYS2:

```
line_test.cpp: In function 'void line_to_comparison(int)':
line_test.cpp:341:21: error: ISO C++11 does not support the 'I' printf flag [-Werror=format=]
             printf( "line_to() executed %d times in %" PRId64 " microseconds.\n",
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
line_test.cpp:341:21: error: format '%d' expects argument of type 'int', but argument 3 has type 'int64_t' {aka 'long long int'} [-Werror=format=]
line_test.cpp:343:21: error: ISO C++11 does not support the 'I' printf flag [-Werror=format=]
             printf( "canonical_line_to() executed %d times in %" PRId64 " microseconds.\n",
                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
line_test.cpp:343:21: error: format '%d' expects argument of type 'int', but argument 3 has type 'int64_t' {aka 'long long int'} [-Werror=format=]
cc1plus.exe: all warnings being treated as errors
make[1]: *** [Makefile:45: objwin/tiles/line_test.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/c/Projects/Cataclysm-DDA/tests'
make: *** [Makefile:1043: tests] Error 2

```

File: tests/map_memory.cpp [hunk 1]
```diff
@@ -102,7 +102,8 @@ TEST_CASE( "lru_cache_perf", "[.]" )
         }
     }
     const auto end1 = std::chrono::high_resolution_clock::now();
-    const long diff1 = std::chrono::duration_cast<std::chrono::microseconds>( end1 - start1 ).count();
+    const int64_t diff1 = std::chrono::duration_cast<std::chrono::microseconds>
+                          ( end1 - start1 ).count();
     printf( "completed %d insertions in %ld microseconds.\n", max_size, diff1 );
```
Comment (ZhilkinSerg) [inline tests/map_memory.cpp]:
```suggestion
    printf( "completed %d insertions in %lld microseconds.\n", max_size, diff1 );
```

**Comment by jbytheway** [comment]
Jenkins rebuild

**Pull request merged**